aptos-move/e2e-move-tests/src/tests/remote_state.rs

```diff
@@ -51,17 +57,28 @@ fn get_account_apt_balance(h: &mut MoveHarness, addr: AccountAddress) -> u64 {
     bcs::from_bytes::<u64>(bytes.as_slice()).unwrap()
 }
 
-/// Simple test that reads the APT balance of the test account.
+/// Reads the APT balance of the test account and checks that it matches the expected value.
 #[ignore]
 #[tokio::test(flavor = "multi_thread")]
 async fn view_existing_account_balance() {
+    // Create a new `MoveHarness` connected to the testnet API endpoint.
+    //
+    // Due to the heavy API usage, you may run into limits if you run remote-state-based
+    // simulations repeatedly or in parallel.
+    //
+    // If that happens, you can create an Aptos Build account and create an API key (for free),
+    // which will increase your rate limit significantly.
+    // - https://build.aptoslabs.com/docs/start#api-quick-start
+    //
+    // To attach your API key here, switch to the alternative constructor
+    // `MoveHarness::new_with_remote_state_with_api_key`.
```

Comment:
I see you added a note here: I think it is still not bad to document the actual APIs ie `new_with_remote_state_with_api_key` vs `new_with_remote_state`, in case tests change, go away, etc.